### ISSUE TITLE
[SilentCrash] When generating Lost Cities space world

### ISSUE THREAD

**Issue by drakray** [MinecraftModDevelopment/OreSpawn#114]
https://pastebin.com/5wnb42jV

```
Description: Exception ticking world

java.lang.NoClassDefFoundError: net/minecraft/world/World$2
```

Space world with SingleBiome on in LostCities, never happened before when creating other world.

**Comment by drakray** [comment]
Tried making another world, same crash https://pastebin.com/RDp35men
Custom profile used https://pastebin.com/zzm6jmNu , named profile_space_empty.cfg, using LostCitesBOP in single player.
Will try another profile, but I made some world earlier and didn't crash, but I wasn't in Creative flying fast.

**Comment by dshadowwolf** [comment]
The last line in the OreSpawn code is a call to "world.setBlockState" - let me go see if @McJty has any ideas on what causes this.

**Comment by drakray** [comment]
Manage to get another [crash](https://pastebin.com/GSqHrwXv) with the Space profile, but with no mention of OreSpawn, just BoP, so it must be on LostCities side.
I didn't get to make it crash with my custom non-sphere profile though.

**Comment by dshadowwolf** [comment]
McJty hasn't responded yet, but I've seen similar crashes in the past where a neighbor change spread quite a bit out - usually accompanied by chunkgen lag warnings

**Comment by drakray** [comment]
Here is a Minutor render of the last map that crashed, I was in the "tube" toward the upper left corner when MC stopped responding, and popped a crash file in the folder, without closing.
![lastcrash](https://user-images.githubusercontent.com/4303126/36650809-5f08fb98-1a73-11e8-80e9-09dadf6565a0.png)

**Comment by drakray** [comment]
As we can see, there is no "chunk branch" in this one, if it's what you are talking about
Sadly I already deleted the logs folder :(

**Comment by McJty** [comment]
It is crashing in oreSpawn as far as I can see. Best to report to them as well

**Comment by dshadowwolf** [comment]
@drakray what version of Lost Cities are you using? It's not in the mod-list provided with the crash log and could have the information needed to figure this out

**Comment by drakray** [comment]
Lost Cities 2.0.5

**Comment by drakray** [comment]
@McJty [this crash](https://pastebin.com/GSqHrwXv) have no mention of Orespawn in it though

**Comment by McJty** [comment]
Yes but neither does it mention Lost Cities

**Comment by drakray** [comment]
yeah I know, that's why I posted it in OreSpawn first, but after a couple of quick test, I only got this in Lost Cities Space profile, not even in "normal" profile :/
Also, it's fairly recent, because couple days/week ago, I created Space map with no problem, first time I got this was yesterday

**Comment by dshadowwolf** [comment]
hrm... common thread in the crashes is that "markAndNotify" is in the chain - is it possible that one of the blocks getting the notification is erroring ?

**Comment by drakray** [comment]
As mentionned on the LostCities Curse page, I tried setting fixVanillaCascading to false(was on true), to no avail
[Here](https://github.com/MinecraftModDevelopmentMods/OreSpawn/files/1759711/LogStuff.zip) is the latest crashLog, with complete log folder, and a Minutor render of the map, I was in the upper right corner when it crashed this time, and it took longer than the last 2 time before crashing.
In fml-junk-earlystartup.log, near line 80350, is the start of crashy info I think

**Comment by dshadowwolf** [comment]
There it is - it got into a recursive loop. Had issues like that early in the OreSpawn 3 development cycle and had to implement a check to make sure that we only ran a given event handler a limited number of times... I think that was it - been a while since I ran into that.

Basically... it's a really, really tricky area of the code and it looks like Lost Cities BOP is getting into a bit of recursion, from the stack trace.

**Comment by josephcsible** [comment]
Any time you get a `java.lang.NoClassDefFoundError: net/minecraft/world/World$2`, the real cause and stack trace is lost. I opened MinecraftForge/MinecraftForge#4769 about this. ~~For debugging, you can mitigate this by calling `Class.forName("net/minecraft/world/World$2")` (and also for any other classes that this happens for) somewhere before things go badly.~~ This is fixed now, so use Forge 2620 or above if you want to see the real stack trace.

**Comment by drakray** [comment]
@josephcsible I tried using 2620 on my crashy settings, but now instead of a silentCrash, the worldgen just stop, I can still move and open inventory, but the chat/command doesn't respond, and if I get far enough from where I was when the worldGen stopped, I can seemingly pass through wall/glass bubble.
No info in [log](https://github.com/MinecraftModDevelopmentMods/OreSpawn/files/1763427/logs.zip), no crash
Lag started at ~22:17, killed the process around ~22:45

**Comment by dshadowwolf** [comment]
We've got a new version on the Maven and this might have been related to a JSON Parse error, in the end. Could you test and see ?

**Comment by drakray** [comment]
[debug.log](https://github.com/MinecraftModDevelopmentMods/OreSpawn/files/1838098/debug.log)
[latest.log](https://github.com/MinecraftModDevelopmentMods/OreSpawn/files/1838099/latest.log)
Log stop at 10:46, killed process at 10:53 because worldGen stopped and couldn't open door/chest/Bauble inventory
w/ latest version of everything on date and forge 2631

**Comment by dshadowwolf** [comment]
Hrm... can you try with the freshly released OreSpawn 3.3.0 and let me know if this still occurs ?

**Comment by dshadowwolf** [comment]
Just did some testing and flying around in a Lost Cities space generation world and have not been able to trigger a crash at all. Closing as resolved in an intervening version.
